DEVELOPER_GUIDE.md

```diff
@@ -78,6 +83,44 @@ Additionally, the `gcc` toolchain needs to be installed on Mac. To install, run:
 brew install gcc
 ```
 
+#### Extra setup for Mac M1 Machines
+
+The following commands enable running/building k-NN on M1 machines:
+
+```bash
+// Go to jni folder
+cd k-NN/jni
+
+// File changes required
+sed -i -e 's/\/usr\/local\/opt\/libomp\//\/opt\/homebrew\/opt\/llvm\//g' CMakeLists.txt
+sed -i -e 's/-march=native/-mcpu=apple-m1/g' external/nmslib/similarity_search/CMakeLists.txt
+sed -i -e 's/pragma message WARN/pragma message /g' external/nmslib/similarity_search/src/distcomp_scalar.cc
+sed -i -e 's/-mcpu=apple-a14/-mcpu=apple-m1/g' external/nmslib/python_bindings/setup.py
+sed -i -e 's/__aarch64__/__undefine_aarch64__/g' external/faiss/faiss/utils/distances_simd.cpp
```

Comment:
have we thought around automating this change as part of gradle build/run?

Reply:
If there is a way to detect specific architecture in the tasks, then we could change that.

Reply:
I think that should be a separate issue though.

Reply:
Make sense. you can resolve this comment, and create a github issue for automating this.

Reply:
https://github.com/opensearch-project/k-NN/issues/1247